File: leetcode/src/main/java/com/giwankim/leetcode/MaxProfit.java
package com.giwankim.leetcode;

public class MaxProfit {
  public int maxProfit(int[] prices) {
    int minPrice = Integer.MAX_VALUE;
    int maxProfit = 0;
    for (int price : prices) {
      maxProfit = Math.max(maxProfit, price - minPrice);
      minPrice = Math.min(price, minPrice);
    }
    return maxProfit;
  }
}

File: leetcode-kt/src/main/kotlin/com/giwankim/leetcode/MaxProfit.kt
package com.giwankim.leetcode

class MaxProfit {
    fun maxProfit(prices: IntArray): Int {
        var maxProfit = 0
        var minPrice = Int.MAX_VALUE
        for (price in prices) {
            maxProfit = maxProfit.coerceAtLeast(price - minPrice)
            minPrice = minPrice.coerceAtMost(price)
        }
        return maxProfit
    }
}
